services/ymax-planner/src/vstorage-utils.ts

```diff
@@ -14,19 +14,40 @@ import { Fail, q } from '@endo/errors';
 // #region vstorage paths, cf. golang/cosmos/x/vstorage/types/path_keys.go
 // TODO: Promote elsewhere, maybe @agoric/internal?
 
+export const EncodedKeySeparator = '\x00';
+export const PathSeparator = '.';
+
+/** @type {(key: string) => string} */
+export const encodedKeyToPath = key => {
+  const encodedParts = key.split(EncodedKeySeparator);
+  encodedParts.length > 1 || Fail`invalid encoded key ${q(key)}`;
+  const path = encodedParts.slice(1).join(PathSeparator);
```

Comment:
This function seems fishy, but I recognise you didn't introduce it in this PR.  I'd encourage looking up its implementation in `path_keys.go` in case it is also fishy.

Namely, here `encodedParts` was split with `EncodedKeySeparator` and is (bizarrely) asserted to have only a single element (`encodedParts[0]`, which is a string), so this is effectively:

```suggestion
  const path = [].join(PathSeparator);
```

Reply:
I think you misread the assertion; `encodedParts` must have a length _greater than_ 1.

Reply:
🤦 Yes, you're right.  Nothing to see here. 😛